### ISSUE TITLE
Maps

### ISSUE THREAD

**Issue by gabrieleborg** [diegogentilepassaro/min_wage_rent#64]
**Goal**
create some graphical visualization of the zipcodes we use, and of the variation in rent and MW.

**Comment by santiagohermo** [comment]
Per meeting, here we want to:

- Make one map of the US that shows all zipcodes in Zillow data
- Find meaningful examples to plot at the MSA level. Ideas: MSAs that border across states, MSAs with some city that has higher MW than the state

**Comment by gabrieleborg** [comment]
Alright guys @santiagohermo @diegogentilepassaro, sorry for the delay but it took me longer than expected to tweak all different details that go into maps..! 

First, I created a main national map showing the location of all zipcodes present in our rent panel (we have a total 3316 zipcodes): 
<img width="893" alt="Screen Shot 2020-11-08 at 1 36 49 PM" src="https://user-images.githubusercontent.com/30894427/98481319-75977200-21c7-11eb-97dd-74e79a493eb9.png">
A minor thing to remember here is that we are dropping from the pic, for clarity reason, 26 zipcodes located in Alaska and Hawaii. 

The most difficult part was to then create "zoomed" maps for specific areas. Here I created, for each area, a set of 3 maps: 
1. a map showing the percentage change in rents
2. a map showing the percentage change in actual MW 
3. a map showing the percentage change in experienced MW

Unfortunately, these map are not very informative I believe. Even if I spent 2 days trying to tweak parameters to make them clearer, I don't think we can clearly see the correlation between rent and MW increases. The main reasons I believe are more than one: first, the estimated rent increase is rather small and this means that it is therefore hard to see it on a small subset of zipodes. Second,  I spent a lot of times trying to find the right city/area to plot, but I noticed that many places where we have a local MW change (either county or city), they also have a simultaneous statewide one. So in these cases, the real variation we are using it's in the intensity of the change, but again, this is very hard to visualize on a small subset of zipcodes from a single area. 

As an example, I show you here the plots for the Seattle MSA: 

**actual MW**
![Seattle_msa_mw](https://user-images.githubusercontent.com/30894427/98481622-8943d800-21c9-11eb-9bb6-37991bfa8dc6.png)

**Experienced MW**
![Seattle_msa_expmw](https://user-images.githubusercontent.com/30894427/98481632-93fe6d00-21c9-11eb-927e-9e6bccc3ae2f.png)

**Rent Change**
![Seattle_msa](https://user-images.githubusercontent.com/30894427/98481650-ad071e00-21c9-11eb-8850-b0efdef0a58e.png)

I am thinking how to improve the visualization of our main variation, because at the moment I honestly don't know whether it makes then sense to include the "zoomed" pictures.

**Comment by santiagohermo** [comment]
Thanks @gabrieleborg. I am definitely going to check your code to learn how to make such beatiful maps!

- The map of the US with the whole set of zipcodes looks really cool. It should definitely be our first appendix figure! I wonder whether it makes sense to color MSAs in a (very) transparent way, just to highlight that we capture urban areas.

- The maps for Seattle look nice too. I agree that it's hard to visualize a correlation, but don't give up yet! A few comments:
   1. I think it would be useful is to mention the period over which we are taking the % change. 
   2. Another potentially useful thing is to use continuous colors with the same "extremes", so that it's easier to see what parts of the map have a high value of the variables.
   3. Maybe we can plot the log difference instead and check? Since this is what we use in the model.

Finally, the maps may be useful to see the minimum wage variable only, and comparing experience vs not experienced. In this sense, I think it would be useful to plot those variables for all zipcodes (since they are well-defined)

**Comment by gabrieleborg** [comment]
@santiagohermo Thanks for the suggestions ! I am working on them now.

**Comment by diegogentilepassaro** [comment]
@gabrieleborg amazing work here! I agree with @santiagohermo that the US map looks amazing and that it should definitely make it to the new draft. A suggestion for that:

- Maybe it is a good idea to create the same US map that has all US zipcodes colored by population densiity or by urban share of population. If we present the map of our zipcodes (hereafter, map 1) next to one of those poeple will really get the sense that we are using really urban zipcodes, which are probably what matters since MW effects are expected there. 

 As for the Seattle map, trying variation in logs as @santiagohermo seems like the way to go!

**Comment by gabrieleborg** [comment]
Thanks for the comment guys @diegogentilepassaro  @santiagohermo . Some updates:

**Sample Map**
I have added a visualization of the Metropolitan Areas in the US. Tomorrow, following Diego's suggestion, I will plot a map showing some population or urban density. 
![sample_map](https://user-images.githubusercontent.com/30894427/98612227-b5428480-22c1-11eb-9b8c-ae1c195867c0.png)

**Zoomed Maps**
Regarding the log-version of the variable, it doesn't change anything: we take the log difference because it approximate well percentage change. I run a safety check doing that and indeed we get basically the same values, so graphically nothing changes. I then apply to modifications: first I switched to a continuous color scale with the same extremes in terms of colors, and I then specified that we are looking at the 6-months percentage change. I don't think things get much clearer: 
**Actual MW**
![Seattle_msa_mw](https://user-images.githubusercontent.com/30894427/98612683-a8726080-22c2-11eb-843f-e01adbafc52d.png)
**Experienced MW**
![Seattle_msa_expmw](https://user-images.githubusercontent.com/30894427/98612705-b1fbc880-22c2-11eb-8593-995f7609ae4c.png)
**Rent change**
![Seattle_msa](https://user-images.githubusercontent.com/30894427/98612721-ba540380-22c2-11eb-8821-85d1192a8a9a.png)

The only nice thing you see is that the experienced MW change actually shows you how the change is smoother than the actual MW suggests. Plots for other cities look similar (you can check them in [`analysis/descriptive/output`](https://github.com/diegogentilepassaro/min_wage_rent/tree/%2364-maps/analysis/descriptive/output)). Tomorrow I will think a bit more at possible improvements, but feel free to suggest!

**Comment by gabrieleborg** [comment]
@diegogentilepassaro @santiagohermo I added a map plotting the population density for metropolitan areas. In this way it's easy to compare it with our initial sample map: 
<img width="669" alt="Screen Shot 2020-11-10 at 3 26 04 PM" src="https://user-images.githubusercontent.com/30894427/98729735-546e8700-2369-11eb-99a5-6b19daf461fb.png">

I think this is a nice comparison, showing how we are indeed capturing the most densely populated zipcodes. I also produced a map with the housing density , but it looks exactly the same. As for the color of the density: I am now using a continuous scale, but the great variance in values makes it somehow "diluted". To reach this result, I already had to winsorize the density values at the 5th and 95th percentiles. If we want more contrast, we will need to use a discrete scale (based on some density distribution's quantiles).
